eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/EventCriteria.java

```diff
@@ -36,124 +36,111 @@
      *
-     * @return a builder that allows criteria to be built matching against any event type.
+     * @param types The types to match against.
+     * @return A new {@code EventCriteria} instance that matches events with the given {@code types}.
      */
-    static EventCriteria.Builder forAnyEventType() {
-        return EventCriteriaBuilder.NO_TYPES;
+    static EventCriteria isOneOfTypes(@Nonnull Set<String> types) {
+        return new EventTypesCriteria(types);
     }
 
     /**
-     * A {@link Set} of {@link String} containing all the types of events applicable for sourcing, streaming, or
-     * appending events.
+     * Construct a new {@code EventCriteria} instance that matches events with the given {@code types}.
      *
-     * @return The {@link Set} of {@link String} containing all the types of events applicable for sourcing, streaming,
-     * or appending events.
+     * @param types The types to match against.
+     * @return A new {@code EventCriteria} instance that matches events with the given {@code types}.
      */
-    Set<String> types();
+    static EventCriteria isOneOfTypes(@Nonnull String... types) {
+        return isOneOfTypes(Set.of(types));
+    }
 
     /**
-     * A {@link Set} of {@link Tag Tags} applicable for sourcing, streaming, or appending events. An {@code Tag} can,
-     * for example, refer to a model's (aggregate) identifier name and value.
+     * Construct a new {@code EventCriteria} instance that matches events with the given {@code types}. All criteria
+     * need to match for the event to be considered a match.
      *
-     * @return The {@link Set} of {@link Tag Tags} applicable for sourcing, streaming, or appending events.
+     * @param criteria The criteria to match against.
+     * @return A new {@code EventCriteria} instance that matches events with the given {@code types}.
      */
-    Set<Tag> tags();
+    static EventCriteria both(EventCriteria... criteria) {
+        return new AndEventCriteria(Arrays.asList(criteria));
+    }
 
     /**
-     * Indicates whether the given {@code type} matches the types defined in this instance. If no types are defined, any
-     * given {@code type} will be considered a match.
+     * Construct a new {@code EventCriteria} instance that matches events with the given {@code types}. All criteria
+     * need to match for the event to be considered a match.
      *
-     * @param type The type to match against this criteria instance.
-     * @return {@code true} if the type matches, otherwise {@code false}.
+     * @param criteria The criteria to match against.
+     * @return A new {@code EventCriteria} instance that matches events with the given {@code types}.
      */
-    default boolean matchingType(String type) {
-        return types().isEmpty() || types().contains(type);
+    static EventCriteria either(EventCriteria... criteria) {
+        return new OrEventCriteria(Arrays.asList(criteria));
```

Comment:
I think it'd be easier to understand if we match methods with class names, so for `OrEventCriteria` use `or`, but as you said in the description it's for discussion. 

IMO `or`/`and` is more popular among developers (either more for functional languages).